src/shared/varlink.c

```diff
@@ -2233,12 +2243,13 @@ static int connect_callback(sd_event_source *source, int fd, uint32_t revents, v
         return 0;
 }
 
-int varlink_server_listen_fd(VarlinkServer *s, int fd) {
-        _cleanup_free_ VarlinkServerSocket *ss = NULL;
+static int varlink_server_create_listen_fd_socket(VarlinkServer *s, int fd, VarlinkServerSocket **ret_ss) {
+        _cleanup_(varlink_server_socket_freep) VarlinkServerSocket *ss = NULL;
         int r;
 
         assert_return(s, -EINVAL);
         assert_return(fd >= 0, -EBADF);
```

Comment:
internal functions should usually use assert(), not assert_return().

assert() → checks for programming errors in *our* code, and is strict and aborts

assert_return() → checks for programming errors in *foreign* code that calls ours, and is more lenient, and just debug logs and returns a error code.

The varlink code is of course currently only used inside of systemd itself, but the goal was to expose it as part of libsystemd.so to other projects, hence it uses assert_return() already.

Hence given this is now an internal function (static), please use simple assert() here, and then move the assert_return() asserts into the outer/non-static varlink_server_listen_fd() instead